### PR TITLE
Add scale decoding of slice of struct.

### LINKED ISSUE #661: scale: decode slices of structs
currently, slices of structs are not supported in scale. DecodeSlice needs to be updated to handle slices of structs.

### PR DESCRIPTION
**Pull request by arijitAD**
## Changes

- Add scale decoding of slice of struct.

## Tests

```
TestEncodeDecodeSliceStruct
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [ ] All integration tests and required coverage checks are passing

## Issues

- closes #661

### CODE REVIEW

File: lib/scale/enc_dec_test.go [hunk 1]
```diff
@@ -189,3 +189,19 @@ func TestEncodeDecode_Array(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, b, dec)
 }
+
+func TestEncodeDecodeSliceStruct(t *testing.T) {
+	type SimpleStruct struct {
+		A int64
+		B bool
+	}
```
Comment (noot) [inline lib/scale/enc_dec_test.go:197]: could you add some more cases, like structs with `[]byte` fields, `common.Hash` fields, maybe fields with custom decode functions also?

**Review by noot — COMMENTED** [review]
looks good! more test cases pls

File: lib/scale/enc_dec_test.go (hunk 1 above)
Comment (arijitAD) [reply lib/scale/enc_dec_test.go:197]: Done

**Review by edwardmack — APPROVED** [review]

**Review by noot — APPROVED** [review]
great stuff :D

**Review by RyRy79261 — APPROVED** [review]

**Review by anthdm — APPROVED** [review]
GJ

**Review by noot — APPROVED** [review]
looks good!

**Pull request merged**